src/main/java/io/confluent/connect/elasticsearch/bulk/BulkProcessor.java

```diff
@@ -341,22 +345,36 @@ private BulkResponse execute() throws Exception {
         log.error("Failed to create bulk request from batch {} of {} records", batchId, batch.size(), e);
         throw e;
       }
-      for (int remainingRetries = maxRetries; true; remainingRetries--) {
+      int retryAttempt = 0;
+      int attempts = 0;
+      while (true) {
         boolean retriable = true;
+        ++attempts;
         try {
           log.trace("Executing batch {} of {} records", batchId, batch.size());
           final BulkResponse bulkRsp = bulkClient.execute(bulkReq);
           if (bulkRsp.isSucceeded()) {
+            if (attempts > 1) {
+              log.debug("Completed batch {} of {} records with attempt {}/{}", batchId, batch.size(), attempts, maxRetries + 1);
+            }
             return bulkRsp;
           }
           retriable = bulkRsp.isRetriable();
           throw new ConnectException("Bulk request failed: " + bulkRsp.getErrorInfo());
         } catch (Exception e) {
-          if (retriable && remainingRetries > 0) {
-            log.warn("Failed to execute batch {} of {} records, retrying after {} ms", batchId, batch.size(), retryBackoffMs, e);
-            time.sleep(retryBackoffMs);
+          if (retriable && retryAttempt < maxRetries) {
+            ++retryAttempt;
+            long sleepTimeMs = computeRetryWaitTimeInMillis(retryAttempt, minRetryBackoffMs, maxRetryBackoffMs);
+            if (log.isDebugEnabled()) {
+              log.debug("Failed to execute batch {} of {} records with attempt {}/{}, will attempt retry after {} ms",
```

Comment:
Feels a bit odd that, depending on the log level, we change what this message is: a WARN or a DEBUG message. Shouldn't it be a WARN always? Should we just print the stack track in a separate debug message?

Reply:
Note that currently the log message outputs the stack trace every time the connector fails and then retries. This was an attempt to change the behavior, but it's more complex than it needs to be.